[tanyqcollin] iP

tanyqcollin opened the pull request:
### Carter
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Carter frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3.  add your tasks.
4. let it manage your tasks for you :kissing_closed_eyes:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadline (coming soon)
- [ ] Reminders (coming soon)

___
If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```
public class Main {
    public static void main(String[] args) {
        new Carter("/data/carter.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

tanyqcollin committed:
rename duke.java, add greeting and exit

tanyqcollin committed:
add echos

tanyqcollin committed:
store text and display text to the user

tanyqcollin committed:
add the ability to mark task as done.

tanyqcollin committed:
support ToDo, Event and Deadlines tasks

tanyqcollin committed:
test ui

tanyqcollin committed:
add error handling

tanyqcollin committed:
support delete task from the list

tanyqcollin committed:
add save data feature

tanyqcollin committed:
Merge branch 'branch-Level-7'

tanyqcollin committed:
able to handle date and time for deadline and event

tanyqcollin committed:
Merge branch 'branch-Level-8'

tanyqcollin committed:
make it more oop

tanyqcollin committed:
create package for each class

tanyqcollin committed:
Merge branch 'add-gradle-support'

tanyqcollin committed:
add gradle

tanyqcollin committed:
add test case for Parser and Deadline

tanyqcollin committed:
change base name of jar

tanyqcollin committed:
add JavaDoc

tanyqcollin committed:
apply coding standards to the files

tanyqcollin committed:
add find keyword feature

tanyqcollin committed:
Merge branch 'branch-A-JavaDoc'

tanyqcollin committed:
Merge branch 'branch-A-CodingStandard'

tanyqcollin committed:
solve merge conflicts

tanyqcollin committed:
fix display issue of the find feature

tanyqcollin committed:
add javadoc to the showMatchingTask

tanyqcollin committed:
add style checking

tanyqcollin committed:
Merge branch 'branch-A-CheckStyle'

rxchell commented on src/main/java/carter/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+     * @return A string representation of the task formatted for file storage.
+     */
+    @Override
+    public String toFileString() {
```
Good naming to differentiate it from the usual toString

rxchell commented on src/main/java/carter/Parser.java:
```diff
@@ -0,0 +1,110 @@
+ */
+public class Parser {
+
+    /**
+     * Parses the user input and performs the corresponding action.
+     *
+     * @param input The user input command as String.
+     * @param tasks The TaskList containing the current tasks.
+     * @param ui The Ui used to display message to the user.
+     * @throws CarterException If the user input is invalid or cannot be recognized.
+     */
```
Good Javadoc comments, complete and adheres to coding guidelines

rxchell commented on src/main/java/carter/Task.java:
```diff
@@ -0,0 +1,77 @@
+     *
+     * @return A string representation of the task formatted for file storage.
+     */
+    public abstract String toFileString();
```
Appropriate use of abstract class based on abstract method

rxchell commented on src/main/java/carter/Ui.java:
```diff
@@ -0,0 +1,118 @@
+ * Represents the user interface of the Carter application.
+ */
+public class Ui {
+    private final static String DIVIDER = "____________________________________________________________";
```
Good use of final and static, and that the constant is in caps.

rxchell commented on src/test/java/carter/ParserTest.java:
```diff
@@ -0,0 +1,95 @@
+
+    @Test
+    public void test() {
+
```
Is the additional line needed?

rxchell reviewed (commented):
LGTM. The code looks good to me in terms of structure and organising the classes. Javadoc comments are present as well, and no major errors observed.

martout2002 commented on src/main/java/carter/Storage.java:
```diff
@@ -0,0 +1,97 @@
+     * @return A list of tasks from storage file.
+     * @throws CarterException If the file is corrupted.
+     */
+    public List<Task> load() throws CarterException {
```
Good javadoc comments, code is easily understandable and i can understand the Storage class easily

martout2002 commented on src/main/java/carter/ToDo.java:
```diff
@@ -0,0 +1,47 @@
+ *  Represents a task with event in Carter
+ */
+public class ToDo extends Task {
+
```
Good use of inheritance in your Task parent class!

martout2002 reviewed (commented):
LGTM you practiced your OO principles well!

tanyqcollin committed:
add GUI

tanyqcollin committed:
Merge branch 'branch-Level-10'

tanyqcollin committed:
apply varargs feature

tanyqcollin committed:
apply varags feature

tanyqcollin committed:
Merge branch 'branch-A-Varargs'

tanyqcollin committed:
Merge branch 'branch-A-Varargs'

tanyqcollin committed:
Add assertions

Add assertions to several method in the Storage, Deadline, Event, Task, TaskList and Response class to ensure critical inputs are not null and index value are within the valid ranges.

These assertions help to catch potential programming error early and improve the overall robustness of the application.

tanyqcollin committed:
Improve parser and storage code to follow Java coding standards

Refactored the Parser class and Storage class by adding utility methods to reduce code duplication.

This comit addresses issue related to code clarity and coding standards compliance, improve the readablity of the codebase.

tanyqcollin committed:
Use stream feature in Response class

tanyqcollin committed:
Merge pull request #1 from tanyqcollin/branch-A-Assertions

Add assertions

tanyqcollin committed:
Merge branch 'master' into branch-A-CodeQuality

tanyqcollin committed:
Merge pull request #3 from tanyqcollin/branch-A-CodeQuality

Improve parser and storage code to follow Java coding standards

tanyqcollin committed:
Merge branch 'master' into branch-A-Streams

tanyqcollin committed:
Merge pull request #4 from tanyqcollin/branch-A-Streams

Use stream feature in Response class

tanyqcollin committed:
remove unused code

tanyqcollin committed:
Add CI to the codebase

tanyqcollin committed:
Add comparable to the task class

tanyqcollin committed:
Merge branch 'branch-C-Sort'

tanyqcollin committed:
Create gradle.yml

tanyqcollin committed:
Remove gradle.yml

tanyqcollin committed:
Merge branch 'master' of https://github.com/tanyqcollin/ip

tanyqcollin committed:
Change codebase to comply the coing style

tanyqcollin committed:
Add test case for Storage class

tanyqcollin committed:
Add archive tasks feature

tanyqcollin committed:
Merge branch 'branch-C-Archive'

tanyqcollin committed:
enhance javadoc comment

tanyqcollin committed:
Able to resize the viewport

tanyqcollin committed:
Enhance the GUI

tanyqcollin committed:
Merge branch 'branch-A-BetterGui'

tanyqcollin committed:
Use Ai to enhance code quality

tanyqcollin committed:
Merge branch 'A-AiAssisted'

tanyqcollin committed:
Add README

tanyqcollin committed:
Merge branch 'branch-A-UserGuide'

tanyqcollin committed:
Fix bug and apply code standard

tanyqcollin committed:
Update readme
